pop/service/api.go

```diff
@@ -0,0 +1,215 @@
+package service
+
+import (
+	"bytes"
+
+	"github.com/BurntSushi/toml"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/base64"
+	"github.com/satori/go.uuid"
+	"gopkg.in/dedis/onet.v1"
+	"gopkg.in/dedis/onet.v1/log"
+	"gopkg.in/dedis/onet.v1/network"
+)
+
+const (
+	// ErrorWrongPIN indicates a wrong PIN
+	ErrorWrongPIN = iota + 4100
+	// ErrorInternal indicates something internally went wrong - see the
+	// error message
+	ErrorInternal
+	// ErrorOtherFinals indicates that one or more of the other conodes
+	// are still missing the finalization-step
+	ErrorOtherFinals
+)
+
+func init() {
+	network.RegisterMessage(&FinalStatement{})
+	network.RegisterMessage(&PopDesc{})
+}
+
+// FinalStatement is the final configuration holding all data necessary
+// for a verifier.
+type FinalStatement struct {
+	// Desc is the description of the pop-party.
+	Desc *PopDesc
+	// Attendees holds a slice of all public keys of the attendees.
+	Attendees []abstract.Point
+	// Signature is created by all conodes responsible for that pop-party
+	Signature []byte
+}
+
+// The toml-structure for (un)marshaling with toml
+type finalStatementToml struct {
+	Desc      *popDescToml
+	Attendees []string
+	Signature string
+}
+
+// NewFinalStatementFromString creates a final statement from a string.
+func NewFinalStatementFromString(s string) *FinalStatement {
```

Comment:
I know returning error is more verbose, but it's how golang does it. The fact that this function does **not** return any errors worries me because it assumes the string is a valid statement. This will probably make something fail later in unexpected situations. I'd gladly see this function returns an error...
Places where things can go wrong:
 + `toml.Decode` at 52
 + `uuid.FromString()` at 59
 + `append(atts, B64ToPoint(p))` at 80: point can be nil and then panic after !
 + `base64.StdEncoding.DecodeString(fsToml.Signature)` line 83